lua_modules/fifo/fifo.lua

```diff
@@ -17,29 +17,33 @@ local tr, ti = table.remove, table.insert
 --
 -- Returns 'true' if the queue contained at least one non-phantom entry,
 -- 'false' otherwise.
-local function dequeue(q,k)
-  if #q > 0
-   then
-     local new, again = k(q[1], #q == 1)
-     if new == nil
-       then tr(q,1)
-            if again then return dequeue(q, k) end -- note tail call
-       else q[1] = new
-     end
-     return true
-   else q._go = true ; return false
+local function dequeue(q, k)
+  if #q > 0 then
+    local new, again = k(q[1], #q == 1)
+    if new == nil then
+      tableRemove(q, 1)
+    else
+      q[1] = new
+    end
+    if again then
+      return dequeue(q, k)
+    end -- note tail call
```

Comment:
I guess I'd never had reason to both replace the top element of the fifo and want to be called back on the thing I just pushed.  That does seem a little odd to want, but I agree that this patch is probably a better thing to do if the `k`onsumer asks for it.

Reply:
I did these changes around two years ago, but the code runs better with it.
IIRC it was that I forced code to send a whole file into the httpserver.lua module including giving the size non chunked but at the beginning and so had some uh, well "interesting" usings of the whole thing.
Should have changed the api instead of hiding it behind specific parametrs, but well.
It just stopped sending in certain cases and I tracked it down to having to return what I just pushed.